app/controllers/thread_controller.php

```diff
@@ -12,36 +18,57 @@ public function index()
         $pagination->checkLastPage($threads);
         $total = Thread::countAll();
         $pages = ceil($total / $per_page);
+        $sub_title = 'Total of '.$total.' threads';
         $this->set(get_defined_vars());
     }
+
     public function view()
     {
-        $thread = Thread::get(Param::get('thread_id'));
-        $thread_info = $thread->getThreadInfo();
-        $thread_id = Param::get('thread_id');
+        if (!isset($_SESSION['username'])) {
+            redirect(url('/'));
+        }
+        $user = new User;
+        $user->setInfoByUsername($_SESSION['username']);
+        try {
+            $thread = Thread::get(Param::get('thread_id', 0));
+        } catch (RecordNotFoundException $e) {
+            $error = true;
+        }
+        if (isset($error)) {
+            $this->set(get_defined_vars());
+            return;
+        }
 
-        $comment_page = Param::get('comment_page',1);
+        $thread_id = Param::get('thread_id', 0);
+        $thread_info = objectToArray($thread->getThreadInfo($thread_id));
+        $comment = new Comment;
+        $comment->id = $thread_id;
+        $comment_page = Param::get('comment_page', 1);
```

Comment:
Please avoid magic numbers :bow: 
Declare this to constant variables instead (including 0).